.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -1,24 +1,61 @@
+---
+.system_probe_file_diff_check:
+  - set +e
+  - |+
+    for path in $SYSTEM_PROBE_SRC_PATH
+    do
+      echo "Checking $path..."
+      echo "$DIFF_FILES" | grep "$path"
+      export exit_code=$?
```

Comment:
Nit: for clarity, I'd give this variable a more specific name, like `file_diff_check_exit_code`. That way, it's easier to read the script of the jobs that use it, as:
```
- if [ ${file_diff_check_exit_code} -eq 0 ]; then
- <script>
- fi
```